raiden-ts/src/transport/epics/webrtc.ts

```diff
@@ -130,117 +138,101 @@ async function getMatrixIceServers(matrix: MatrixClient): Promise<RTCIceServer[]
   return servers;
 }
 
-type ExtMatrixEvent = MatrixEvent & {
-  getContent: () => { msgtype: string; body: string };
-};
-
 // returns a stream of filtered, valid MatrixEvents
 function matrixWebrtcEvents$<T extends RtcEventType>(
-  matrix: MatrixClient,
+  action$: Observable<RaidenAction>,
   type: T,
-  sender: string,
-  { caps }: Pick<RaidenConfig, 'caps'>,
+  sender: Address,
+  callId?: string,
 ) {
-  return merge(
-    fromEvent<[MatrixEvent]>(matrix, 'Room.timeline').pipe(pluck(0)),
-    getCap(caps, Capabilities.TO_DEVICE) ? fromEvent<MatrixEvent>(matrix, 'toDeviceEvent') : EMPTY,
-  ).pipe(
+  return action$.pipe(
+    filter(messageReceived.is),
     filter(
-      (event): event is ExtMatrixEvent =>
-        event.getType() === 'm.room.message' &&
-        event.getSender() === sender &&
-        (event as ExtMatrixEvent).getContent()?.msgtype === 'm.notice',
+      (action) => action.meta.address === sender && action.payload.msgtype === rtcMatrixMsgType,
     ),
-    mergeMap((event) => {
+    mergeMap(function* (action) {
       try {
-        return of(decode(rtcCodecs[type], jsonParse(event.getContent()?.body)));
+        yield decode(rtcCodecs[type], jsonParse(action.payload.text));
       } catch (e) {}
```

Comment:
Should we at least log something here in case of an error?

Reply:
This isn't critical, but I think it'd be good, in case this starts to fail silently